PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskSVtaskMCFilter.cxx

```diff
@@ -0,0 +1,191 @@
+fFilteredTracksArray(0x0),
+fFilteredTracksName("mytracks"),
+fAodEvent(0x0),
+fMCHeader(0x0),
+fMCPartArray(0x0)
+{
+   //constructor
+}
+//_______________________________________________________________________________
+AliAnalysisTaskSVtaskMCFilter::~AliAnalysisTaskSVtaskMCFilter()
+{
+   //destructor
+}
+//_______________________________________________________________________________
+void AliAnalysisTaskSVtaskMCFilter::UserCreateOutputObjects(){
+ //dummy
+  return;
+}
+//_______________________________________________________________________________
+void AliAnalysisTaskSVtaskMCFilter::ExecOnce()
+{
+
+   //
+   // To be executed only once, for the first event
+   //
+
+   AliDebug(2, "Entering ExecOnce()");
+
+   // Load the event
+   fAodEvent = dynamic_cast<AliAODEvent*>(fInputEvent);
+
+   if(!fAodEvent){
+       if (AODEvent() && IsStandardAOD()) {
+
+          // In case there is an AOD handler writing a standard AOD, use the AOD
+          // event in memory rather than the input (ESD) event.
+          fAodEvent = dynamic_cast<AliAODEvent*>(AODEvent());
+       }
+   } 
+
+   fMCPartArray = dynamic_cast<TClonesArray*>(fAodEvent->FindListObject(AliAODMCParticle::StdBranchName()));
+   if(!fMCPartArray) {
+      AliError(Form("MC particles not found! Task '%s' will be disabled!", GetName()));
+      return;
+   }
+
+   fMCHeader = (AliAODMCHeader*)fAodEvent->GetList()->FindObject(AliAODMCHeader::StdBranchName());
+   if(!fMCHeader) {
+      AliError(Form("MC header not found! Task '%s' will be disabled!", GetName()));
+      return;
+   }
+
+  // add the filtered tracks to event if not yet there
+  if (!(InputEvent()->FindListObject(fFilteredTracksName))) {
+     fFilteredTracksArray = new TClonesArray("AliAODTrack");
+     fFilteredTracksArray->SetName(fFilteredTracksName);
+     ::Info("AliAnalysisTaskSVtaskMCFilter::ExecOnce", "track collection with name '%s' has been added to the event.", fFilteredTracksName.Data());
+     InputEvent()->AddObject(fFilteredTracksArray);
+  }
+  else {
+    AliError(Form("%s: Object with name %s already in event! Returning", GetName(), fFilteredTracksName.Data()));
+    return;
+  }
+  
+   AliAnalysisTaskEmcal::ExecOnce();
+}
+
+//_______________________________________________________________________________
+Bool_t AliAnalysisTaskSVtaskMCFilter::Run()
+{
+
+   //
+   // Analysis execution
+   //
+
+    //fFilteredTracksArray->Delete();
+    fFilteredTracksArray->Clear();
+
+   // fix for temporary bug in ESDfilter
+   // the AODs with null vertex pointer didn't pass the PhysSel
+   if (!fAodEvent->GetPrimaryVertex() || TMath::Abs(fAodEvent->GetMagneticField()) < 0.001) return kFALSE;
+
+ 
+
+   GetTrackContainer(0)->ResetCurrentID();
+   AliAODTrack* track = 0;
+   TString nameGen;
+   Int_t lab, mother;
+   AliAODMCParticle *mcpart = NULL;
+   Int_t n = fFilteredTracksArray->GetEntriesFast();
+
+   while ((track = static_cast<AliAODTrack*>(GetTrackContainer(0)->GetNextAcceptParticle()))) {
```

Comment:
As noted [here](http://alidoc.cern.ch/AliPhysics/master/class_ali_particle_container.html#a6363a52ba754afdf0c4c357a281de6f1), this approach is deprecated because it can cause problems with multiple particle or track containers. Please update to using the iterator approach. See [here](http://alidoc.cern.ch/AliPhysics/master/READMEcontainers.html#emcalContainerIterateTechniques) for more information.